src/components/switch/switch.tsx

```diff
@@ -0,0 +1,37 @@
+import { BaseComponent } from "../common/base-component";
+import { Component, Event, EventEmitter, Prop } from "@stencil/core";
+import { SwitchRender } from "../renders/bootstrap/switch/switch-render";
+@Component({
+  host: {
+    role: "switch"
+  },
+  shadow: false,
+  styleUrl: "switch.scss",
+  tag: "gx-switch"
+})
+export class Switch extends SwitchRender(BaseComponent) {
+  /**
+   * Attribute that provides the caption to the <label> tag.
+   */
+  @Prop() caption: string;
+  /**
+   * Indicates that the checked is selected by default.
+   */
+  @Prop({ mutable: true })
+  checked: boolean;
+  /**
+   * This attribute lets specify if the element is disabled.
+   */
+  @Prop() disabled = false;
+  /**
+   * The 'change' event is emitted when a change to the element's value is committed by the user.
+   */
+  @Event() onChange: EventEmitter;
+
+  /**
```

Comment:
Take a look at the `id` property of `<gx-checkbox>`. I'd copy/paste the comment here.

Reply:
The comment looks very similar to ```<gx-checkbox>``` control. Also explains how control ```id``` value works through all the control. Explains wht happens if ou assign an ```id``` to the control or not.
Shall I remove that? i think that comment is useful, don't u think?

Reply:
The comment looks very similar to ```<gx-checkbox>``` control. But also explains how control ```id``` value works through all the control. Explains wht happens if u assign an ```id``` to the control or not.
Shall I remove that? i think that comment is useful, don't u think?

Reply:
I think the fact that there's an input and a label element is hidden to the user that reads this doc, so there's no much value explaining that and could lead to confusion.

Reply:
Got it!